src/final/hooks/use-prefetch-proposals.ts

```diff
@@ -60,20 +73,27 @@ export const usePrefetchProposals = ({ widgetData }: Properties) => {
   const activeSources = useMemo(() => {
     const sources: ProposalSource[] = [];
 
-    if (hasAgoraProposal) {
+    if (hasAgoraProposal && agoraEnabled) {
       sources.push('agora');
     }
 
-    if (hasSnapshotProposal) {
+    if (hasSnapshotProposal && snapshotEnabled) {
       sources.push('snapshot');
     }
 
-    if (hasTallyProposal) {
+    if (hasTallyProposal && tallyEnabled) {
       sources.push('tally');
     }
```

Comment:
hm I think we can remove that 2nd condition since queries are disabled if given feature is disabled :think: but not sure what if we have some proposals and after that we change value in settings, is it the reason you've added it?

Reply:
exactly, otherwise the widgets do not disappear when changing the settings after the proposals have already been fetched